src/main/drivers/vtx_common.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* Created by jflyper */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <ctype.h>
+#include <string.h>
+
+#include "platform.h"
+#include "build/debug.h"
+
+#if defined(VTX_COMMON)
+
+#include "vtx_common.h"
+
+vtxDevice_t *vtxDevice = NULL;
+
+void vtxCommonInit(void)
+{
+}
+
+// Whatever registered last will win
+
+void vtxCommonRegisterDevice(vtxDevice_t *pDevice)
+{
+    vtxDevice = pDevice;
+}
+
+void vtxCommonProcess(uint32_t currentTimeUs)
+{
+    if (!vtxDevice)
+        return;
+
+    if (vtxDevice->vTable->process)
+        vtxDevice->vTable->process(currentTimeUs);
+}
+
+vtxDevType_e vtxCommonGetDeviceType(void)
+{
+    if (!vtxDevice)
+        return VTXDEV_UNKNOWN;
+
+    return vtxDevice->devtype;
+}
+
+// band and chan are 1 origin
+void vtxCommonSetBandChan(uint8_t band, uint8_t chan)
```

Comment:
Did you consider using 'parameter' abstraction? Most of these functions do similar thing, it may be better to define enum with parameter ids and then 'vtxCommonSetParameterInt(vtxParam_e id, int value);` 'int vtxCommonGetParameterInt(vtxParam_e id);`

Using distinct function may get out of hand easily ..